File: M3Tools/Controls/People/Users/LoginFields.vb
Public Class LoginFields
    ''' <summary>
    ''' Gets or sets the username
    ''' </summary>
    ''' <returns></returns>
    Property Username As String
        Get
			Return If(String.IsNullOrWhiteSpace(uf_Username.Username), Nothing, uf_Username.Username)
		End Get
        Set(value As String)
            uf_Username.Username = value
        End Set
    End Property

    ''' <summary>
    ''' Gets or sets the password
    ''' </summary>
    ''' <returns></returns>
    Property Password As String
        Get
			Return If(String.IsNullOrWhiteSpace(pf_Password.Password), Nothing, pf_Password.Password)
		End Get
        Set(value As String)
            pf_Password.Password = value
        End Set
    End Property

    ReadOnly Property UsernameField As UsernameField
        Get
            Return uf_Username
        End Get
    End Property

	ReadOnly Property PasswordField As PasswordField
		Get
			Return pf_Password
		End Get
	End Property

	''' <summary>
	''' Clears all text from the username and password fields
	''' </summary>
	Public Sub Clear()
		ClearUsername()
		ClearPassword()
	End Sub

	Public Sub ClearUsername()
		uf_Username.Clear()
	End Sub

	Public Sub ClearPassword()
		pf_Password.Clear()
	End Sub

	Public Shadows Function Focus(Optional field As String = "u") As Boolean
		Select Case field
			Case "u"
				Return UsernameField.Focus()
			Case "p"
				Return PasswordField.Focus()
		End Select
	End Function

	Private Sub PasswordGotFocus(sender As Object, e As EventArgs) Handles pf_Password.PasswordGotFocus
		PasswordField.SelectAll()
	End Sub
End Class

File: M3Tools/Controls/People/Users/LoginFields.cs
using System;

namespace SPPBC.M3Tools
{
	/// <summary>
	/// 
	/// </summary>
	public enum Field
	{
		/// <summary>
		/// The username field
		/// </summary>
		Username,
		/// <summary>
		/// The password field
		/// </summary>
		Password,
		/// <summary>
		/// Both fields
		/// </summary>
		Both
	}

	/// <summary>
	/// 
	/// </summary>
    public partial class LoginFields
    {
        /// <summary>
    /// Gets or sets the username
    /// </summary>
    /// <returns></returns>
        public string Username
        {
            get
            {
                return string.IsNullOrWhiteSpace(uf_Username.Username) ? string.Empty : uf_Username.Username;
            }
            set
            {
                uf_Username.Username = value;
            }
        }

        /// <summary>
    /// Gets or sets the password
    /// </summary>
    /// <returns></returns>
        public string Password
        {
            get
            {
                return string.IsNullOrWhiteSpace(pf_Password.Password) ? string.Empty : pf_Password.Password;
            }
            set
            {
                pf_Password.Password = value;
            }
        }

		/// <summary>
		/// 
		/// </summary>
        public LoginFields()
        {
            InitializeComponent();
        }

        /// <summary>
	/// 	''' Clears all text from the username and password fields
	/// 	''' </summary>
        public void Clear(Field field = Field.Both)
        {
			switch (field)
			{
				case Field.Username:
					uf_Username.Clear();
					break;
				case Field.Password:
					pf_Password.Clear();
					break;
				default:
					uf_Username.Clear();
					pf_Password.Clear();
					break;
			}
        }

		/// <summary>
		/// Focus on one of the fields
		/// </summary>
		/// <param name="field"></param>
		/// <returns></returns>
        public bool Focus(Field @field = Field.Username)
        {
            switch (@field)
            {
                case Field.Username: return uf_Username.Focus();
                case Field.Password: return pf_Password.Focus();
				case Field.Both:
				default: throw new ArgumentException($"'${field}' is not a valid value");
            }
        }

        private void PasswordGotFocus(object sender, EventArgs e)
        {
            pf_Password.SelectAll();
        }
    }
}
